### PR TITLE
feature(cli): integrate Expo Atlas in `expo start` and `expo export`

### PR DESCRIPTION
**Pull request by byCedric**
# Why

This is an experimental branch that I'm using to see how well Expo Atlas (and the information it provides) works for local development.

Currently, the feature is put behind a feature flag `EXPO_UNSTABLE_ATLAS=true`, and only works for `expo start` and `expo export`. Once we are confident this works well, we can add a CLI flag like `--atlas` or `--stats`, and update the output overview to include the `.expo/atlas.jsonl` file and how to open it (it's `npx expo-atlas` within project, or `npx expo-atlas ./path/to/atlas.jsonl`).

> [!NOTE]
> Funfact: Expo Atlas `0.1.0` is running Expo SDK 51 (`expo@51.0.0-preview.0`).

# How

- Added `expo-atlas@^0.1.0` as dev dependency to `@expo/cli`
- Added `AtlasPrerequisite` to ensure `expo-atlas@^0.1.0` is installed in project
- Added `attachAtlas` to **src/start/server/metro/instantiateMetro.ts**

# Test Plan

**For exporting**
- `$ bun create expo ./test-atlas --template tabs`
- `$ cd ./test-atlas`
- `$ EXPO_UNSTABLE_ATLAS=true expod export ...`
- `$ npx expo-atlas`

**For development mode**
- `$ bun create expo ./test-atlas --template tabs`
- `$ cd ./test-atlas`
- `$ EXPO_UNSTABLE_ATLAS=true expod start`
- `$ open http://localhost:8081/_expo/atlas` ([link](http://localhost:8081/_expo/atlas))

**Try it out in NCL** 🚀
- `$ cd ./apps/native-component-list`
- `$ EXPO_UNSTABLE_ATLAS=true expod start`
- `$ open http://localhost:8081/_expo/atlas` ([link](http://localhost:8081/_expo/atlas))

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/@expo/cli/package.json [hunk 1]
```diff
@@ -155,6 +155,7 @@
     "@types/wrap-ansi": "^8.0.1",
     "@types/ws": "^8.5.4",
     "devtools-protocol": "^0.0.1113120",
+    "expo-atlas": "0.0.14",
```
Comment (byCedric) [inline packages/@expo/cli/package.json:158]: Hardcoded for now, need to answer these questions:
  - Do we want to keep optionally ship Atlas -- right now it's optional, only activated when installed
  - How do we want to ensure the version match -- for now, it's just a `typeof require('expo-atlas/cli').createExpoAtlasMiddleware === 'function'` check.

Comment (byCedric) [reply packages/@expo/cli/package.json]: We are shipping atlas as a separate library, but with integration inside the Expo CLI. See the `AtlasPrerequisite` and `attachAtlasAsync` functions.

**Review by amandeepmittal — APPROVED** [review]
Docs changes are 👍

File: packages/@expo/cli/src/start/server/metro/debugging/AtlasPrerequisite.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  private async ensureAtlasInstalled(options: Partial<EnsureDependenciesOptions> = {}) {
+    try {
+      return await ensureDependenciesAsync(this.projectRoot, {
+        ...options,
+        installMessage:
+          'Expo Atlas is required to gather bundle information, but it is not installed in this project.',
+        warningMessage:
+          'Expo Atlas is not installed in this project, unable to gather bundle information.',
+        requiredPackages: [
+          { version: '^0.1.0', pkg: 'expo-atlas', file: 'expo-atlas/package.json', dev: true },
```
Comment (byCedric) [inline packages/@expo/cli/src/start/server/metro/debugging/AtlasPrerequisite.ts:29]:
```suggestion
          { version: '^0.1.1', pkg: 'expo-atlas', file: 'expo-atlas/package.json', dev: true },
```

**Comment by byCedric** [comment]
Hmm, interesting. I think my computer can't properly regenerate the lockfile for some reason. Now, the e2e CLI tests are passing...

**Pull request merged**